test/unit/test-mysqlDao.js

```diff
@@ -1898,4 +1898,112 @@ describe('MysqlDao', function() {
+
+            return userDao.getAll({})
+                .then(users => {
+                    return topSecretInfoDao.leftJoin(users, 'secretAgentCode')
+                })
+                .then(users => {
+                    Should.exist(users);
+                    users.should.have.lengthOf(3);
+                    users[0].should.have.property('id', 1);
+                    users[0].should.have.property('topSecretInfo');
+                    users[0].topSecretInfo.should.have.lengthOf(2);
+                    users[0].topSecretInfo[0].should.have.property('secretAgentCode', 1);
+                    users[0].topSecretInfo[0].should.have.property('intel', "It's gonna rain.");
+                    users[0].topSecretInfo[1].should.have.property('intel', "The sky is blue.");
+
+                    users[1].should.have.property('id', 2);
+                    users[1].should.have.property('topSecretInfo');
+                    users[1].topSecretInfo.should.have.lengthOf(0);
+
+                    users[2].should.have.property('id', 3);
+                    users[2].should.have.property('topSecretInfo');
+                    users[2].topSecretInfo.should.have.lengthOf(500);
+                })
+        });
+
+        it('Should join target entities to a single source entity', function() {
+            return userDao.getOneById(1)
+                .then(user => {
+                    return topSecretInfoDao.leftJoin(user, 'secretAgentCode')
+                })
+                .then(user => {
+                    Should.exist(user);
+                    user.should.have.property('id', 1);
+                    user.should.have.property('topSecretInfo');
+                    user.topSecretInfo.should.have.lengthOf(2);
+                })
+        });
+
+        it('Should join target entities using a different target join key and joined entities key name', function() {
+            return userDao.getAll({})
+                .then(users => {
+                    return topSecretInfoDao.leftJoin(users, 'handlerCode', {joinKey: 'id', resultsKey: 'other_joined_entities_key_name'})
+                })
+                .then(users => {
+                    Should.exist(users);
+                    users.should.have.lengthOf(3);
+
+                    users[0].should.have.property('id', 1);
+                    users[0].should.have.property('other_joined_entities_key_name');
+                    users[0].other_joined_entities_key_name.should.have.lengthOf(0);
+
+                    users[1].should.have.property('id', 2);
+                    users[1].should.have.property('other_joined_entities_key_name');
+                    users[1].other_joined_entities_key_name.should.have.lengthOf(1);
+                    users[1].other_joined_entities_key_name[0].should.have.property('intel', "Hello world.");
+                })
+        });
+
+        it('Should join in batches', function() {
```

Comment:
Just for the sake of completeness, can you add a test for a getAll that returns no results